bridges/OpenlyBridge.php

```diff
@@ -0,0 +1,247 @@
+			$url = $this->getURI();
+		}
+
+		$html = getSimpleHTMLDOM($url);
+		$html = defaultLinkTo($html, $this->getURI());
+
+		if ($html->find('h1', 0)) {
+			$this->feedTitle = $html->find('h1', 0)->plaintext;
+		}
+
+		if ($html->find('h2.title-v4', 0)) {
+			$html->find('span.tooltiptext', 0)->innertext = '';
+			$this->feedTitle = $html->find('a.tooltipitem', 0)->plaintext;
+		}
+
+		foreach($html->find('div.item') as $div) {
+			$this->items[] = $this->getArticle($div->find('a', 0)->href);
+
+			if (count($this->items) >= $this->itemLimit) {
+				break;
+			}
+		}
+	}
+
+	public function getURI() {
+		switch ($this->queriedContext) {
+			case 'All News':
+				return self::URI . 'news';
+				break;
+			case 'All Opinion':
+				return self::URI . 'people';
+				break;
+			case 'By Tag':
+				return self::URI . $this->getInput('content') . '/?theme=' . $this->getInput('tag');
+			case 'By Region':
+				return self::URI . $this->getInput('content') . '/?region=' . $this->getInput('region');
+				break;
+			case 'By Author':
+				return self::URI . 'profile/?id=' . $this->getInput('profileId');
+				break;
+			default:
+				return parent::getURI();
+		}
+	}
+
+	public function getName() {
+		switch ($this->queriedContext) {
+			case 'All News':
+				return 'News - Openly';
+				break;
+			case 'All Opinion':
+				return 'Opinion - Openly';
+				break;
+			case 'By Tag':
+				if (empty($this->feedTitle)) {
+					$this->feedTitle = $this->getInput('tag');
+				}
+
+				if ($this->getInput('content') === 'people') {
+					return $this->feedTitle . ' - Opinion - Openly';
+				}
+
+				return $this->feedTitle . ' - Openly';
+				break;
+			case 'By Region':
+				if (empty($this->feedTitle)) {
+					$this->feedTitle = $this->getInput('region');
+				}
+
+				if ($this->getInput('content') === 'people') {
+					return $this->feedTitle . ' - Opinion - Openly';
+				}
+
+				return $this->feedTitle . ' - Openly';
+				break;
+			case 'By Author':
+				if (empty($this->feedTitle)) {
+					$this->feedTitle = $this->getInput('profileId');
+				}
+
+				return $this->feedTitle . ' - Author - Openly';
+				break;
+			default:
+				return parent::getName();
+		}
+	}
+
+	private function getAjaxURI() {
+		$part = '/ajax.html?';
+
+		switch ($this->queriedContext) {
+			case 'All News':
+				return self::URI . 'news' . $part;
+				break;
+			case 'All Opinion':
+				return self::URI . 'people' . $part;
+				break;
+			case 'By Tag':
+				return self::URI . $this->getInput('content') . $part . 'theme=' . $this->getInput('tag');
+				break;
+			case 'By Region':
+				return self::URI . $this->getInput('content') . $part . 'region=' . $this->getInput('region');
+				break;
+		}
+	}
+
+	private function getArticle($url) {
+		$article = getSimpleHTMLDOMCached($url, self::ARTICLE_CACHE_TIMEOUT);
+		$article = defaultLinkTo($article, $this->getURI());
+
+		$article->find('span.lead-text', 0)->outertext = ''; // Remove lead text
```

Comment:
"Warning: Creating default object from empty value in /var/www/rss-bridge/pr2129/bridges/OpenlyBridge.php on line 228"

https://feed.eugenemolotov.ru/pr2129/?action=display&bridge=Openly&context=By+Region&region=north-america&content=news&format=Html